From 9c8e2ea2767de36ef0eec2c57fdd248c3fc65cdf Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 03 Oct 2022 20:21:18 +0300
Subject: [PATCH] [SW-106502] gaudi3: add pcie ctrl check from rr config

security_rr_close_pcie_config function to close range register
is called from both dies based on RR fuse bit as part of security
init. This patch adds pcie_ctrl fuse check for configuring RR
for slave die which allows RR configuration on slave die only
if pci_ctrl bit is set to 1.

Change-Id: I1911a6f454d0566497744d6656a8eb5f81fead99
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 3c899e0..6b3000a 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -15,6 +15,7 @@
 #include <gaudi3_security.h>
 #include <gaudi3/gaudi3.h>
 #include <gaudi3_blocks.h>
+#include <dev_common.h>
 
 #define RR_LBW_PRIV_RANGE_MIN_SHORT_VAL		0x0
 #define RR_LBW_PRIV_RANGE_MAX_SHORT_VAL		0x1FFFF000
@@ -702,6 +703,11 @@
 
 void security_rr_close_pcie_config(void)
 {
+	uint32_t pcie_ctrl_val = arcboot_rom_fuse_pcie_ctrl_config();
+
+	if (arcboot_rom_global.die_id && !pcie_ctrl_val)
+		return;
+
 	security_rr_config_pcie_lbw(&rr_ctx_lbw[RR_CLOSE_ALL]);
 
 	security_rr_config_pcie_hbw(&rr_ctx_lbw[RR_CLOSE_ALL]);
